.github/workflows/user-branch-rules.yml

```diff
@@ -0,0 +1,33 @@
+name: User Branch Rules
+
+permissions:
+  contents: read
+
+
+on:
+  push:
+    branches:
+      - 'users/**'
+
+
+jobs:
+  user-branch-rules:
+    if: github.repository_owner == 'llvm'
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check branch length
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+          MAX_BRANCH_LENGTH: 200
```

Comment:
200 seems too high. Git convention is 80 characters for the subject, users//spr/ is 11 characters, GitHub usernames are limited to 39 characters, that gives 130 characters.

(spr can suffix the commit message with a -1 etc when there are conflicts, but that's rare, and there's enough slack in the above in practice that we can ignore all that, e.g, usernames are likely way less than 39 characters)

Reply:
Importantly:

.git/logs/refs/remotes/origin/ is 30 chars, plus NUL, plus <letter>:\ is 34 chars. That leaves just 26 for the path to the checkout if you have a branch that's 200 chars long, but Users\Michael\llvm-project hits that limit exactly (as the first name of the right length to come into my head).

Reply:
refs/heads/ is also included in the ref name, so that adds another 11 characters.  Does spr but the commit subject in the branch name?

Reply:
Yes: https://github.com/getcord/spr/blob/1e74fa165da4f4ee626ec8ddbb8414c7f29464be/spr/src/config.rs#L91-L100

Reply:
Related https://github.com/getcord/spr/issues/186

Reply:
Nope; see the linked issue

Reply:
But we already have "The title should be concise. Because all commits are emailed to the list with the first line as the subject, long titles are frowned upon. Short titles also look better in git log." in the developer policy, so this *shouldn't* be an issue if people actually adhere to our policy... (ha)

Reply:
80-100 should be sane upper limit, yes.